plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -187,122 +130,62 @@ def process_main(self):
 
-        :return:    a unit instance
-        :rtype:     pulp.plugins.model.Unit
-        """
-        model = models.Blob(unit_dict['digest'])
-        return self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, {},
-                                            model.relative_path)
 
-
-class GetLocalManifestsStep(GetLocalUnitsStep):
+class SaveUnitsStep(publish_step.SaveUnitsStep):
     """
-    Get the manifests we have locally and ensure that they are associated with the repository.
+    Save the Units that need to be added to the repository and move them to the content folder.
     """
-    def _dict_to_unit(self, unit_dict):
+    def __init__(self):
         """
-        convert a unit dictionary (a flat dict that has all unit key, metadata,
-        etc. keys at the root level) into a Unit object. This requires knowing
-        not just what fields are part of the unit key, but also how to derive
-        the storage path.
-
-        Any keys in the "metadata" dict on the returned unit will overwrite the
-        corresponding values that are currently saved in the unit's metadata. In
-        this case, we pass an empty dict, because we don't want to make changes.
-
-        :param unit_dict:   a flat dictionary that has all unit key, metadata,
-                            etc. keys at the root level, representing a unit
-                            in pulp
-        :type  unit_dict:   dict
-
-        :return:    a unit instance
-        :rtype:     pulp.plugins.model.Unit
+        Initialize the step, setting its description.
         """
-        model = self.parent.parent.step_get_metadata.manifests[unit_dict['digest']]
-        return self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, model.metadata,
-                                            model.relative_path)
-
+        super(SaveUnitsStep, self).__init__(step_type=constants.SYNC_STEP_SAVE)
+        self.description = _('Saving Manifests and Blobs')
 
-class SaveUnitsStep(PluginStep):
-    def __init__(self, working_dir):
+    def get_iterator(self):
         """
-        :param working_dir: full path to the directory into which blob files
-                            are downloaded. This directory should contain one
-                            directory for each docker blob, with the ID of the
-                            docker blob as its name.
-        :type  working_dir: basestring
+        Return an iterator that will traverse the list of Units that were downloaded.
+
+        :return: An iterable containing the Blobs and Manifests that were downloaded and are new to
+                 Pulp.
+        :rtype:  iterator
         """
-        super(SaveUnitsStep, self).__init__(
-            step_type=constants.SYNC_STEP_SAVE, plugin_type=constants.IMPORTER_TYPE_ID,
-            working_dir=working_dir)
-        self.description = _('Saving manifests and blobs')
+        return iter(self.parent.step_get_local_manifests.units_to_download +
```

Comment:
If you want to avoid making a new list (I assume these two data structures are lists or similar), you can instead use ``itertools.chain`` to produce the same behavior.